### PR TITLE
Clean up global Mat Vec class interface

### PR DESCRIPTION
**Pull request by norihiro-w**
follow up of #1246 

moved all linear algebraic operations in global mat vec classes to BLAS namespace. The next step is renaming "BLAS" to something else.

### CODE REVIEW

**Comment by chleh** [comment]
Before going on we should coordinate things, because I aslo have a private branch with 1000+ lines of changes in matrix/vector classes.

**Comment by norihiro-w** [comment]
@chleh what have you changed? My changes are as already explained in #1167

**Comment by chleh** [comment]
This PR is OK, but before renaming BLAS, please let's talk at first.
Actually I removed nearly all methods from the global matrix and vector classes. Maybe that's not the best solution. But I thought, if algebra is done by stand-alone functions, reading and writing could also be done by those.

File: MathLib/LinAlg/UnifiedMatrixSetters.cpp [hunk 1]
```diff
@@ -89,9 +89,27 @@ void setVector(Eigen::VectorXd& v, MatrixVectorTraits<Eigen::VectorXd>::Index co
 namespace MathLib
 {
 
-double norm(PETScVector const& x)
+double norm(PETScVector const& x, MathLib::VecNormType nmtype = MathLib::VecNormType::NORM2)
```
Comment (chleh) [inline MathLib/LinAlg/UnifiedMatrixSetters.cpp]: Suggest to remove this method from this file (maybe in a later PR).
I am against default arguments. They only make the code more error-prone.
I have some code regarding norms in a private branch. I'll make a PR in the next few weeks.

File: Tests/MathLib/TestGlobalMatrixInterface.cpp [hunk 2]
```diff
@@ -94,24 +94,12 @@ void checkGlobalMatrixInterfaceMPI(T_MATRIX &m, T_VECTOR &v)
 
     MathLib::finalizeMatrixAssembly(m);
 
-    // Multiply by a vector
-    v = 1.;
-    const bool deep_copy = false;
-    T_VECTOR y(v, deep_copy);
-    m.multiply(v, y);
-
-    ASSERT_EQ(sqrt(3*(3*3 + 7*7)), y.getNorm());
```
Comment (chleh) [inline Tests/MathLib/TestGlobalMatrixInterface.cpp]: Shall we better keep the test and rewrite it in terms of `BLAS::`? Or shall we remove it entirely?

File: MeshLib/IO/VtkIO/VtuInterface.cpp [hunk 3]
```diff
@@ -86,7 +86,7 @@ bool VtuInterface::writeToFile(std::string const &file_name)
                                        + std::to_string(mpi_rank) + ".vtu";
     const bool vtu_status_i = writeVTU<vtkXMLUnstructuredGridWriter>(file_name_rank);
     bool vtu_status = false;
-    MPI_Allreduce(&vtu_status_i, &vtu_status, 1, MPI_C_BOOL, MPI_LAND, PETSC_COMM_WORLD);
+    MPI_Allreduce(const_cast<bool*>(&vtu_status_i), &vtu_status, 1, MPI_C_BOOL, MPI_LAND, PETSC_COMM_WORLD);
```
Comment (chleh) [inline MeshLib/IO/VtkIO/VtuInterface.cpp]: I think it would be better to declare `vtu_status_i` non-const above.

**Comment by chleh** [comment]
after questions have been answered :+1:

File: Tests/MathLib/TestGlobalMatrixInterface.cpp (hunk 2 above)
Comment (norihiro-w) [inline Tests/MathLib/TestGlobalMatrixInterface.cpp]: These tests have been reactivated using BLAS

File: MathLib/LinAlg/UnifiedMatrixSetters.cpp (hunk 1 above)
Comment (norihiro-w) [inline MathLib/LinAlg/UnifiedMatrixSetters.cpp]: changing the behavior is not a part of this PR, though I agree having a default value here is no help.

**Comment by TomFischer** [comment]
Looks good. :fast_forward:

File: MathLib/LinAlg/UnifiedMatrixSetters.cpp (hunk 1 above)
Comment (chleh) [inline MathLib/LinAlg/UnifiedMatrixSetters.cpp]: Sorry, I missed that.

File: Tests/MathLib/TestGlobalMatrixInterface.cpp (hunk 2 above)
Comment (chleh) [inline Tests/MathLib/TestGlobalMatrixInterface.cpp]: Sorry, I missed that.

**Pull request merged**